dvc/repo/reproduce.py

```diff
@@ -31,6 +31,7 @@ def _reproduce_stage(stages, node, force, dry, interactive, no_commit):
     return [stage]
 
 
+@slow_link_guard
```

Comment:
I don't think we should set it for every method. Your initial approach with decorating `link` is correct. It is especially correct if you think about that individual link time matters and not the overall time like you are capturing in your current PR. You just need to make it print that warning once. Maybe a flag var somewhere or something like that?